src/algorithm.rs

```diff
@@ -344,44 +391,33 @@ mod tests {
             )
         }
 
-        let mock_ledger = MockLedger::default();
-        let set_osrank = Box::new(|node: &Artifact<String>, rank| match node.data() {
-            ArtifactType::Project { osrank: _ } => ArtifactType::Project { osrank: rank },
-            ArtifactType::Account { osrank: _ } => ArtifactType::Account { osrank: rank },
-        });
-
         assert_eq!(network.edge_count(), 11);
 
-        // This is the insertion point of the Graph API. If we had a GraphAPI
-        // "handle" in scope here, we could extract the seed from some state
-        // and use it in the algorithm. Faking it for now.
-
+        // NOTE(adn) In the "real world" the initial_seed will be provided
+        // by who calls `.execute`, probably the ledger/protocol.
         let initial_seed = [0; 16];
 
-        assert_eq!(
-            osrank_naive::<MockLedger, MockNetwork, XorShiftRng>(
-                Some(seed_set),
-                &mut network,
-                &mock_ledger,
-                initial_seed,
-                set_osrank
-            )
-            .unwrap(),
-            ()
-        );
+        let algo: Mock<OsrankNaiveAlgorithm<MockNetwork, MockLedger>> = Mock {
+            unmock: OsrankNaiveAlgorithm::default(),
+        };
+
+        let mut ctx = OsrankNaiveMockContext::default();
+        ctx.seed_set = Some(&seed_set);
+
+        assert_eq!(algo.execute(&mut ctx, &mut network, initial_seed), Ok(()));
+
         assert_eq!(
             network.nodes().fold(Vec::new(), |mut ranks, node| {
-                // let bla = *node.data();
                 ranks.push(format!("{}", *node));
                 ranks
             }),
             vec![
-                "id: p1 osrank: 0.1425",
-                "id: p2 osrank: 0.2225",
-                "id: p3 osrank: 0.1575",
-                "id: a1 osrank: 0.08",
-                "id: a2 osrank: 0.23",
-                "id: a3 osrank: 0.055",
+                "id: p1 osrank: 0.1075",
+                "id: p2 osrank: 0.2325",
+                "id: p3 osrank: 0.2125",
+                "id: a1 osrank: 0.0575",
+                "id: a2 osrank: 0.2675",
+                "id: a3 osrank: 0.065",
```

Comment:
This is because now we are not _recreating_ the RNG every time (remember we had this discussion already?) but rather we are using the same _initial seed_ for the whole algorithm, this is why (I believe) the ranks now change.